### PR TITLE
Fix some lint warnings

### PR DESCRIPTION
**Pull request by cristaloleg**
Have fixed few warnings. I haven't touched this one:
```
frame.go:181:13:warning: struct of size 24 could be 16 (maligned)
```
I'm not sure how it's safe to change order of fields for `Header`(this might also impact `Frame`)

### CODE REVIEW

**Comment by cristaloleg** [comment]
hmm, can we drop support for go 1.7 ? It's not supported anymore.
If you agree on this I'll also update `.travis.yml`

File: dialer.go [hunk 1]
```diff
@@ -526,7 +526,7 @@ func matchSelectedExtensions(selected []byte, wanted, received []httphead.Option
 // connection "poisoned" by SetDeadline() call. In that case done(&err) will
 // store at *err ctx.Err() result. If err is caused not by timeout, it will
 // leaved untouched.
-func setupContextDeadliner(conn net.Conn, ctx context.Context) (done func(*error)) {
+func setupContextDeadliner(ctx context.Context, conn net.Conn) (done func(*error)) {
```
Comment (gobwas) [inline dialer.go]: Position of args was chosen consciously – context has no impact on the operation of `setupContextDeadliner` – it controls the lifetime of `conn`.

Comment (gobwas) [reply dialer.go]: It should be interpreted like if it will be the `time.Duration` type or `<-chan struct{}`.

**Comment by gobwas** [comment]
FYI: reduced `Header` size by your advice, thanks )

File: dialer.go (hunk 1 above)
Comment (cristaloleg) [reply dialer.go]: I did this reorder just because it is written in go best practices

**Comment by cristaloleg** [comment]
My proposal regarding drop support for 1.7 is based only at eol of support, not because there is 2 redundant method calls :)
I’ll address your comments next week(hopefully this PR isn’t so urgent)

**Comment by cristaloleg** [comment]
@gobwas kindly ping 😉

**Comment by gobwas** [comment]
@cristaloleg there are two pending comments – one is for `1.x` version in the travis.ci file and second is my request to revert changes in `setupContextDeadliner` function signature =)

**Comment by gobwas** [comment]
Should be `1.x` actually be `1.10.x`? ) Currently I've just added `"1.10"`.

**Comment by cristaloleg** [comment]
Regarding travis version. There is a `1.x` versions that represents the latest release: https://github.com/travis-ci/travis-build/blob/master/public/version-aliases/go.json#L4
Regarding `setupContextDeadliner` I have misunderstood you. I thought you've accepted this change. will redo in a minute :)

File: .travis.yml [hunk 2]
```diff
@@ -5,9 +5,10 @@ os:
   - osx
 
 go:
-  - 1.7
-  - 1.8
-  - 1.9
+  - 1.7.x
+  - 1.8.x
+  - 1.9.x
+  - 1.x
```
[the remark this answers is not in the document]
Comment (cristaloleg) [reply .travis.yml]: `1.x` stand fot `1.10.x`(https://github.com/travis-ci/travis-build/blob/master/public/version-aliases/go.json#L4)
Should we pin specifically to `1.10` ?

**Comment by gobwas** [comment]
Thanks! 👍

**Pull request merged**